strip-nondeterminism: add page (#4647)
tldr-pages/tldr: pages/common/strip-nondeterminism.md

# strip-nondeterminism

> A tool to remove non-deterministic information (e.g. timestamps) from files.

> 

Reviewer: ```suggestion
> More information: <https://salsa.debian.org/reproducible-builds/strip-nondeterminism>.

```

> - Strip non-deterministic information from a file:

Reviewer: Based on the `README` is with no `-`:
```suggestion
- Strip nondeterministic information from a file:
```

> `strip-nondeterminism {{file}}`

Reviewer: ```suggestion
`strip-nondeterminism {{path/to/file}}`
```

> - Strip nondeterministic information from a file while manually specifying the file type:

Reviewer: ```suggestion
- Strip nondeterministic information from a file manually specifying the filetype:
```

> `strip-nondeterminism --type {{ar|gzip|jar|zip}} {{file}}`

Reviewer: If `ar|gzip|jar|zip` is not the only supported file types you should use a variable:
```suggestion
`strip-nondeterminism --type {{filetype}} {{path/to/file}}`
```

Author: I thought that they were the only ones, because the manpage only lists these four types. It turns out that there are indeed more: https://salsa.debian.org/reproducible-builds/strip-nondeterminism/-/blob/master/lib/File/StripNondeterminism.pm#L116

- Strip nondeterministic information from a file; instead of removing timestamps set them to the specified UNIX timestamp:

> `strip-nondeterminism --timestamp {{unix-timestamp}} {{file}}`

Reviewer: Use `_`:
```suggestion
`strip-nondeterminism --timestamp {{unix_timestamp}} {{path/to/file}}`
```
